lib/ecto_mnesia/planner.ex

```diff
@@ -66,10 +66,15 @@ defmodule EctoMnesia.Planner do
       "Selecting all records by match specification `#{inspect(match_spec)}` with limit #{inspect(limit)}"
     end)
 
+    # still dont know where to get fields from
+    # as they are not coming on the parameters of this function
+    fields = []
+    mapper = processor(preprocess, fields, sources)
```

Comment:
I believe we don't need to preprocess fields at all, it is required only to keep backward compatibility with older Ecto versions and we can live without it. I would try to make it work without backwards compat first.